openlibrary/plugins/importapi/code.py

```diff
@@ -63,8 +63,10 @@ def parse_data(data):
     :rtype: (dict|None, str|None)
     :return: (Edition record, format (rdf|opds|marcxml|json|marc)) or (None, None)
     """
+    if six.py3 and isinstance(data, bytes):
+        data = data.decode('utf-8')
```

Comment:
This is incorrect. Binary MARC, which `data` could be is not utf-8 encoded.

Parse_data must be able to accept bytes (binary MARC) as well as the other more string-like formats (JSON / XML) etc.

Reply:
What is really meant by [`:param str data: Raw data`](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/importapi/code.py#L62)?  Is `data` a `str` or is it `bytes` or do we have to be ready for both?

It sounds like we need to accept both.  If `data` is bytes, how do we know which encoding to use?  hint: We can't guess.

Perhaps we need an issue to clear up [other similar instances](https://github.com/internetarchive/openlibrary/search?p=1&q=%3Aparam+str+data%3A+Raw+data) with real (ugly) type hints.
```python
from typing import Dict, Optional, Tuple, Union
def parse_data(data: Union[bytes, str]) -> Tuple[Optional[Dict], Optional[str]]:
```